http-03-get-and-transform-data/src/app/app.component.ts

```diff
@@ -10,12 +12,33 @@ import { Product } from './model/product.model';
 })
 export class AppComponent implements OnInit {
   loadedPosts: Product[] = [];
-  savedPosts: Product[] = [];
+  startedEditing = new Subject<number>();
+  @ViewChild('postForm') productForm: NgForm; 
+  subscription: Subscription;
+  editMode = false;
+  editItemIndex: number;  
+  editedItem: Product;
+  currentId: number; 
 
   constructor(private http: HttpClient) {}
+  
 
   ngOnInit() {
-    this.fetchPosts();
+    this.subscription = this.startedEditing
+      .subscribe(
+        (index: number) => {
+          this.editItemIndex = index;
+          console.log(this.getProduct(index));
+          this.editMode = true;
+          this.editedItem = this.getProduct(index);
+          this.productForm.setValue({
+            naam: this.editedItem.naam,
```

Comment:
We also need to set here a product-id, otherwise API doesn't know which one to edit:

id: this.getProduct(index).id,

But before, a product Id should be also added in a template:
        <div class="form-group" >
          <label for="id">Id</label>
          <input
            type="text"
            class="form-control"
            id="id"
            ngModel
            readonly
            name="id"
          />
        </div>